buildpipeline/Core-Setup-Linux-Arm-BT.json

```diff
@@ -356,7 +356,7 @@
       "allowOverride": true
     },
     "DockerCommonRunArgs": {
-      "value": "--name $(PB_DockerContainerName) -v \"$(PB_SourcesDirectory):$(PB_GitDirectory)\" -w=\"$(PB_GitDirectory)\" $(PB_DockerImageName)"
+      "value": "--name $(PB_DockerContainerName) -v \"$(PB_SourcesDirectory):$(PB_GitDirectory)\" -w=\"$(PB_GitDirectory)\" -e \"PACKAGEVERSIONPROPSURL=$(PB_PackageVersionPropsUrl)\" $(PB_DockerImageName)"
```

Comment:
Dumb question - why is this using an environment variable and not passing the value in to the MSBuild properties?

Reply:
Short answer: the package version props download happens during init-tools (https://github.com/dotnet/buildtools/pull/1787).

MSBuild reentrancy would have been neat, but there were enough complications that I decided not to do it for this initial implementation. I think I wrote them down somewhere, but I can't find it now. Top things that stick out are that there doesn't seem to be a way to "soft cancel" the requested target (to download, instead), and I couldn't find any reasonable way to pass down *all* properties into the inner layer.

Reply:
We already have places that generate .props files during the build.

https://github.com/dotnet/buildtools/blob/78260239e4db68ad5c2df774a27d5f9a54fd92e5/src/Microsoft.DotNet.Build.Tasks/PackageFiles/BuildVersion.targets#L6
https://github.com/dotnet/buildtools/blob/03988127f489bcd1cf58d71131ac628620d53de9/src/Microsoft.DotNet.Build.Tasks/GenerateConfigurationProps.cs#L15

I'm not sure I understand why this couldn't have been done similarly.  Downloading a props file from the internet, and generating it on the fly, work the same way.  The values aren't available in the root `build.proj` project.  But for all the projects that get invoked from it, the values are available.

Reply:
Maybe that would work, I don't know if we need the values in the root build.proj. Filed https://github.com/dotnet/buildtools/issues/1813.